x/gov/client/utils/query.go

```diff
@@ -54,24 +53,8 @@ func QueryDepositsByTxQuery(clientCtx client.Context, params v1.QueryProposalPar
 		deposits = append(deposits, initialDeposit)
 	}
 
-	searchResult, err := combineEvents(
-		clientCtx, defaultPage,
-		// Query legacy Msgs event action
-		[]string{
-			fmt.Sprintf("%s.%s='%s'", sdk.EventTypeMessage, sdk.AttributeKeyAction, v1beta1.TypeMsgDeposit),
-			fmt.Sprintf("%s.%s='%d'", types.EventTypeProposalDeposit, types.AttributeKeyProposalID, params.ProposalID),
-		},
-		// Query proto Msgs event action v1beta1
-		[]string{
-			fmt.Sprintf("%s.%s='%s'", sdk.EventTypeMessage, sdk.AttributeKeyAction, sdk.MsgTypeURL(&v1beta1.MsgDeposit{})),
-			fmt.Sprintf("%s.%s='%d'", types.EventTypeProposalDeposit, types.AttributeKeyProposalID, params.ProposalID),
-		},
-		// Query proto Msgs event action v1
-		[]string{
-			fmt.Sprintf("%s.%s='%s'", sdk.EventTypeMessage, sdk.AttributeKeyAction, sdk.MsgTypeURL(&v1.MsgDeposit{})),
-			fmt.Sprintf("%s.%s='%d'", types.EventTypeProposalDeposit, types.AttributeKeyProposalID, params.ProposalID),
-		},
-	)
+	q := fmt.Sprintf("%s.%s='%d'", types.EventTypeProposalDeposit, types.AttributeKeyProposalID, params.ProposalID)
```

Comment:
As explained by @jmalicevic, CometBFT changed its logic for querying events.
Because we emit two different events (one containing the message action (from baseapp) and one containing these events keys) this was returning nothing. The new logic wants all the conditions to be valid in the same event.
I've checked that at no other place in the gov module these event keys were emitted with the same attributes in another message. We can then simplify the logic.

I believe, however, we should include that somewhere in our docs (or CometBFT's docs).